compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/calls/Arguments.kt

```diff
@@ -371,10 +375,18 @@ private fun checkApplicabilityForArgumentType(
             return type
         }
 
+        // Reaching here means argument types mismatch, and we want to record whether it's due to the nullability by checking a subtype
+        // relation with nullable expected type.
+        val isMismatchDueToNullability = AbstractTypeChecker.isSubtypeOf(
```

Comment:
May be it's worth checking first that `argumentType` is nullable and `actualExpectedType` is not null, and call `isSubtypeOf` only in case both these conditions are true.

Reply:
Good point! Thanks for the suggestion!